unleash/Unleash.swift

```diff
@@ -35,12 +35,31 @@ public class Unleash {
         self.refreshInterval = refreshInterval
         self.strategies = strategies
         
+        register()
+    }
+    
+    private func register() {
         let body: ClientRegistration = ClientRegistration(appName: appName, strategies: strategies)
-        registerService.register(url: URL(string: self.url)!, body: body)
-            .done {_ in
-            }.catch {error in
-                print("error \(error)")
+        attempt(maximumRetryCount: 3, delayBeforeRetry: .seconds(60)) {
+            self.registerService.register(url: URL(string: self.url)!, body: body)
+        }.done {_ in
+        }.catch {error in
+            debugPrint("error \(error)")
+        }
+    }
+    
+    private func attempt<T>(maximumRetryCount: Int,
+                            delayBeforeRetry: DispatchTimeInterval,
+                            _ body: @escaping () -> Promise<T>) -> Promise<T> {
```

Comment:
I didn't know this random `@escaping()` thing that means nothing to me until Github syntax highlighting put it in another color.

Reply:
This is so it gets held in memory so it can be executed at a later point. By default, Swift 5 is non-escaping.